protos/authorization.proto

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// -----------------------------------------------------------------------------
+
+syntax = "proto3";
+option java_multiple_files = true;
+option java_package = "sawtooth.sdk.protobuf";
+option go_package = "authorization_pb2";
+
+message ConnectionRequest {
+	string endpoint = 1;
+}
+
+enum RoleType {
+	// A shorthand request for asking for all allowed roles.
+	ALL = 0;
+	// Role defining validator to validator communication
+  NETWORK = 1;
+}
+
+message ConnectionResponse {
+  enum Status {
+      OK = 0;
+      ERROR = 1;
+  }
+
+	enum RoleType {
+		// A shorthand request for asking for all allowed roles.
+		ALL = 0;
+		// Role defining validator to validator communication
+	  NETWORK = 1;
+	}
+
+		//Authorization Type required for the authorization procedure
+	  enum AuthorizationType {
+	  	TRUST = 0;
+	  	CHALLENGE = 1;
+	  }
+
+	message RoleEntry {
+		RoleType role = 1;
+		AuthorizationType auth_type = 2;
+	}
+
+	repeated RoleEntry roles = 1;
+  Status status = 2;
+}
+
+message AuthorizationTrustRequest {
+	enum RoleType {
+		// A shorthand request for asking for all allowed roles.
+		ALL = 0;
+		// Role defining validator to validator communication
+		NETWORK = 1;
+	}
+	// A set of requested RoleTypes
+	repeated RoleType requested_roles = 1;
```

Comment:
Since `requested_roles` and `approved_roles` below are both `repeated RoleType`, they would be easier to work with if they were both just called `roles`.